index.js

```diff
@@ -162,9 +168,9 @@ function closeStream (payload) {
   }
 }
 
-function getEncodingHeader (supportedEncodings, request) {
-  const header = request.headers['accept-encoding']
-  return encodingNegotiator.negotiate(header, supportedEncodings)
+function getEncodingHeader (encodings, request) {
+  const header = (request.headers['accept-encoding'] || '').replace('*', 'gzip')
```

Comment:
can you please explain this?

Reply:
Ah, yeah. That was to preserve the existing behavior where a client accepting `*` was interpreted as `gzip` — otherwise it sends Brotli with this patch. I wasn’t sure whether that was behavior I should preserve or not.

Reply:
Great, would you mind adding a comment?

Reply:
I think it would be better to to change line 40 & line 43 back to use push and remove this.
It chooses the first encoding of the encoding list if a client sends `*`.

Reply:
If 40/43 use `.push`, #61 happens. I’m not against your suggestion, but it seemed odd for the compression plugin for a server ecosystem focused on speed to not use the best encoding by default.

Reply:
I am fine with changing the preferred encoding to `br`.
But if we do so we should also return `br` for `*` and change the test accordingly.

Reply:
I’m not, gzip has been widely supported for a decade at least, while brotli is a new thing. I would like to be conservative. I’m happy to have an option to change this.

Reply:
That was my thinking as well. I’m not against a `bleedingEdgeEncodingAssumption` (or whatever), but I’m not certain the benefit outweighs the possible breakage and added one more source of configuration entropy.

Reply:
> I’m not, gzip has been widely supported for a decade at least, while brotli is a new thing. I would like to be conservative. I’m happy to have an option to change this.

That makes sense to me.

> If 40/43 use `.push`, #61 happens. I’m not against your suggestion, but it seemed odd for the compression plugin for a server ecosystem focused on speed to not use the best encoding by default.

Might you please add a test with `Accept-Encoding: gzip, deflate, br` as mentioned in #61. This test would still return `gzip`. 
After thinking a bit about this I noticed that the problem from #61 remains with the mentioned header since the `encoding-negotiator` module sorts the encodings from the header by the quality values, which in this case are all 1 and then returns the first matching supported encoding.
I wrote a (failing) test to show that: https://github.com/SerayaEryn/encoding-negotiator/pull/10/files
Because of that a change to the `encoding-negotiator` module is required. I can do that tomorrow.

The changes to the lines 42, 43 & 172 however are not necessary.

Reply:
I opened a PR in the `encoding-negotiator` module, that adds a `prefferedEncoding` option: https://github.com/SerayaEryn/encoding-negotiator/pull/10
Please let me know if this solution looks good to you.

Reply:
I wonder if an array of encodings in preference order would be more robust (e.g. you ideally want Brotli, but maybe the Node you’re running on doesn’t support it. Then, it would fall back to the next encoding in the array.)

The other nit is I believe it [should be spelled](https://en.wiktionary.org/wiki/preferred) `preferredEncoding`.